src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/ExpandableStringEnumDiagnosticRule.java

```diff
@@ -0,0 +1,69 @@
+package com.azure.tools.apiview.processor.diagnostics.rules;
+
+import com.azure.tools.apiview.processor.diagnostics.DiagnosticRule;
+import com.azure.tools.apiview.processor.model.APIListing;
+import com.azure.tools.apiview.processor.model.Diagnostic;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getClassName;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getPublicOrProtectedMethods;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.makeId;
+import static com.azure.tools.apiview.processor.model.DiagnosticKind.ERROR;
+import static com.azure.tools.apiview.processor.model.DiagnosticKind.WARNING;
+
+/**
+ * Diagnostic rule to validate if types extending from ExpandableStringEnum include
+ * fromString() and values() methods.
+ */
+public class ExpandableStringEnumDiagnosticRule implements DiagnosticRule {
+
+    @Override
+    public void scanIndividual(final CompilationUnit cu, final APIListing listing) {
+
+        cu.getPrimaryType().ifPresent(typeDeclaration -> {
+            if (typeDeclaration instanceof ClassOrInterfaceDeclaration) {
+                ClassOrInterfaceDeclaration classDeclaration = (ClassOrInterfaceDeclaration) typeDeclaration;
+                // check if the type extends ExpandableStringEnum
+                if (classDeclaration.getExtendedTypes() != null
+                        && classDeclaration.getExtendedTypes().stream()
+                        .anyMatch(extendedType -> extendedType.getNameAsString().equals("ExpandableStringEnum"))) {
+                    checkRequiredMethodsExist(cu, listing);
+                }
+            }
+        });
```

Comment:
At some point it might be cool to have a `SubTypeDiagnosticRule` where the constructor just takes `ExpandableStringEnum` (or the appropriate parent type name), and then calls into the abstract method to process when it matches. Not saying we should do it now, but it would be nice to avoid this code block every time.

Reply:
Yeah, I was considering this but didn't have any other subtype rule now, so, didn't add it but will be a good refactoring to do when we have more than one diagnostic rule requiring this.